### PR TITLE
Master onclose history back lpe

### PR DESCRIPTION
**Pull request by kebeclibre**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by kebeclibre** [comment]
robodoo rebase-ff

**Comment by kebeclibre** [comment]
@svs-odoo FYI

File: addons/web/static/src/webclient/actions/action_service.js [hunk 1]
```diff
@@ -536,9 +536,10 @@ function makeActionManager(env) {
                     dialogCloseResolve();
                 }
             }
-            onHistoryBack() {
+            async onHistoryBack() {
                 const previousController = controllerStack[controllerStack.length - 2];
-                if (previousController && !dialogCloseProm) {
+                if (previousController && !dialog.remove) {
```
Comment (aab-odoo) [inline addons/web/static/src/webclient/actions/action_service.js:541]: I think we should set dialog to null when there's no dialog

File: addons/web/static/src/webclient/actions/action_service.js [hunk 2]
```diff
@@ -536,9 +536,10 @@ function makeActionManager(env) {
...
+                    await dialogCloseProm;
```
Comment (aab-odoo) [inline addons/web/static/src/webclient/actions/action_service.js:542]: not sure we should wait for it, as if we do, we go back to B for a brief moment. If we don't wait, we may cancel the rendering of B and directly go back to A, isn't it?

File: addons/web/static/src/webclient/actions/action_service.js [hunk 3]
```diff
@@ -572,10 +572,11 @@ function makeActionManager(env) {
             const removeDialog = env.services.dialog.add(ActionDialog, actionDialogProps, {
                 onClose: () => {
                     if (dialog.remove) {
-                        if (dialog.onClose) {
-                            dialog.onClose();
-                        }
+                        const { onClose } = dialog;
                         dialog = {};
+                        if (onClose) {
+                            onClose({ special: true });
```
Comment (aab-odoo) [inline addons/web/static/src/webclient/actions/action_service.js:579]: what are the specs around this param actually?

**Review by aab-odoo — COMMENTED** [review]
2 cool and small commits :-)
I left a few comments, we can talk about them

File: addons/web/static/src/webclient/actions/action_service.js (hunk 2 above)
Comment (kebeclibre) [reply addons/web/static/src/webclient/actions/action_service.js:542]: fair enough

File: addons/web/static/src/webclient/actions/action_service.js (hunk 3 above)
Comment (kebeclibre) [reply addons/web/static/src/webclient/actions/action_service.js:579]: I don't know really. In legacy (14.0) ; dialog closed that way are not given anything (the on_close callback is passed jqueryEvent.info which seems undefined everytime).

I drew from the action_service:@DoActionButton where when we hit a "cancel" button we pass special to the callback. I don't think a "cancel" button is in any way different from the cross at the top right of a dialog....

**Comment by kebeclibre** [comment]
@aab-odoo refactoring to set dialog to null when absent is done

**Comment by aab-odoo** [comment]
robodoo rebase-ff r+

**Pull request closed without merging**